File: app/src/main/java/com/lucasurbas/counter/app/di/rx/AndroidInteractorSchedulers.java
package com.lucasurbas.counter.app.di.rx;

import javax.inject.Inject;

import io.reactivex.Scheduler;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

public class AndroidInteractorSchedulers implements InteractorSchedulers {

    @Inject
    public AndroidInteractorSchedulers() {
    }

    @Override
    public Scheduler getBackgroundScheduler() {
        return Schedulers.io();
    }

    @Override
    public Scheduler getMainThreadScheduler() {
        return AndroidSchedulers.mainThread();
    }
}

File: app/src/main/java/com/lucasurbas/counter/app/di/rx/AndroidInteractorSchedulers.kt
package com.lucasurbas.counter.app.di.rx

import io.reactivex.Scheduler
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers
import javax.inject.Inject

class AndroidInteractorSchedulers @Inject constructor() : InteractorSchedulers {

    override val backgroundScheduler: Scheduler
        get() = Schedulers.io()

    override val mainThreadScheduler: Scheduler
        get() = AndroidSchedulers.mainThread()

    override val singleScheduler: Scheduler
        get() = Schedulers.single()
}
